itest/litd_mode_integrated_test.go

```diff
@@ -2,24 +2,279 @@ package itest
+		)
+	}
+	loopMacaroonFn = func(cfg *NodeConfig) string {
+		return cfg.LoopMacPath
+	}
+	poolRequestFn = func(ctx context.Context,
+		c grpc.ClientConnInterface) (proto.Message, error) {
+
+		poolConn := poolrpc.NewTraderClient(c)
+		return poolConn.GetInfo(
+			ctx, &poolrpc.GetInfoRequest{},
+		)
+	}
+	poolMacaroonFn = func(cfg *NodeConfig) string {
+		return cfg.PoolMacPath
+	}
+
+	endpoints = []struct {
+		name           string
+		macaroonFn     macaroonFn
+		requestFn      requestFn
+		successPattern string
+	}{{
+		name:           "lnrpc",
+		macaroonFn:     lndMacaroonFn,
+		requestFn:      lndRequestFn,
+		successPattern: "\"identity_pubkey\":\"0",
+	}, {
+		name:           "frdrpc",
+		macaroonFn:     faradayMacaroonFn,
+		requestFn:      faradayRequestFn,
+		successPattern: "\"reports\":[]",
+	}, {
+		name:           "looprpc",
+		macaroonFn:     loopMacaroonFn,
+		requestFn:      loopRequestFn,
+		successPattern: "\"swaps\":[]",
+	}, {
+		name:           "poolrpc",
+		macaroonFn:     poolMacaroonFn,
+		requestFn:      poolRequestFn,
+		successPattern: "\"accounts_active\":0",
+	}}
 )
 
 // testModeIntegrated makes sure that in integrated mode all daemons work
 // correctly.
 func testModeIntegrated(net *NetworkHarness, t *harnessTest) {
 	ctx := context.Background()
 
-	time.Sleep(2 * time.Second)
-
+	// Some very basic functionality tests to make sure lnd is working fine
+	// in integrated mode.
 	net.SendCoins(t.t, btcutil.SatoshiPerBitcoin, net.Alice)
 
+	// We expect a non-empty alias (truncated node ID) to be returned.
 	resp, err := net.Alice.GetInfo(ctx, &lnrpc.GetInfoRequest{})
 	require.NoError(t.t, err)
-
 	require.NotEmpty(t.t, resp.Alias)
+	require.Contains(t.t, resp.Alias, "0")
+
+	t.t.Run("certificate check", func(tt *testing.T) {
+		runCertificateCheck(tt, net.Alice)
+	})
+	t.t.Run("gRPC macaroon auth check", func(tt *testing.T) {
+		cfg := net.Alice.Cfg
+
+		for _, endpoint := range endpoints {
+			endpoint := endpoint
+			tt.Run(endpoint.name+" lnd port", func(ttt *testing.T) {
+				runGRPCAuthTest(
+					tt, cfg.RPCAddr(), cfg.TLSCertPath,
+					endpoint.macaroonFn(cfg),
+					endpoint.requestFn,
+					endpoint.successPattern,
+				)
+			})
+
+			tt.Run(endpoint.name+" lit port", func(ttt *testing.T) {
+				runGRPCAuthTest(
+					tt, cfg.LitAddr(), cfg.TLSCertPath,
+					endpoint.macaroonFn(cfg),
+					endpoint.requestFn,
+					endpoint.successPattern,
+				)
+			})
+		}
+	})
+}
+
+// runCertificateCheck checks that the TLS certificates presented to clients are
+// what we expect them to be.
+func runCertificateCheck(t *testing.T, node *HarnessNode) {
+	// In integrated mode we expect the LiT HTTPS port (8443 by default) and
+	// lnd's RPC port to present the same certificate, namely lnd's TLS
+	// cert.
+	litCerts, err := getServerCertificates(node.Cfg.LitAddr())
+	require.NoError(t, err)
+	require.Len(t, litCerts, 1)
+	require.Equal(
+		t, "lnd autogenerated cert", litCerts[0].Issuer.Organization[0],
```

Comment:
Code was copied over for cert generation so it retains the "lnd" prefix here?

Reply:
No, in integrated mode lnd's certificate is used for both ports. In remote mode litd's certificate is used.